lib/friendly_shipping/services/ship_engine/bad_request.rb

```diff
@@ -0,0 +1,25 @@
+require 'json'
+
+module FriendlyShipping
+  module Services
+    class ShipEngine
+      class BadRequest < StandardError
+        attr_reader :rest_error, :response
+
+        def initialize(rest_error)
+          @rest_error = rest_error
+          @response = rest_error.response
+          super parse_json_errors
+        end
+
+        private
+
+        def parse_json_errors
+          parsed_body = JSON.parse(response.body)
+          messages = parsed_body['errors'].map { |e| e['message'] }
```

Comment:
Potential `Undefined method .map for NilClass` here. As we parse this in a error object itself, we must be sure that this code is very safe. Maybe dealing with `JSON::ParseError`  as well?